aws/resource_aws_instance.go

```diff
@@ -843,13 +843,13 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.Set("ebs_optimized", instance.EbsOptimized)
-	if instance.SubnetId != nil && *instance.SubnetId != "" {
+	if instance.SubnetId != nil && aws.StringValue(instance.SubnetId) != "" {
```

Comment:
`aws.StringValue` converts a `nil` input to `""`, so `if aws.StringValue(instance.SubnetId) != ""` should be sufficient here.

Reply:
ahh of course 😅, will update